espnet/asr/asrtts_utils.py

```diff
@@ -0,0 +1,905 @@
+#!/usr/bin/env python
+
+# Copyright 2017 Johns Hopkins University (Shinji Watanabe)
+#  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
+
+import copy
+import itertools
+import json
+import logging
+import math
+# matplotlib related
+import os
+import shutil
+import tempfile
+
+# chainer related
+import chainer
+
+from chainer import training
+from chainer.training import extension
+
+from chainer.serializers.npz import DictionarySerializer
+from chainer.serializers.npz import NpzDeserializer
+
+# io related
+import kaldi_io_py
+import matplotlib
+import numpy as np
+import torch
+matplotlib.use('Agg')
+
+
+# * -------------------- training iterator related -------------------- *
+def make_batchset_asr(data, batch_size, max_length_in, max_length_out,
```

Comment:
Check `espnet.utils` not to redefine these functions. I'm afraid this makes us hard to maintain and test.
For example,

- make_batchset_xxx -> espnet.utils.training.batchfy
- LoadInputsAndOutputs -> espnet.utils.io_utils
- other -> espnet.asr.asr_utils, etc

https://github.com/espnet/espnet/tree/v.0.4.0/espnet/utils